sqlmesh/core/engine_adapter/shared.py

```diff
@@ -192,6 +192,29 @@ def is_multi_catalog_supported(self) -> bool:
         return self.is_requires_set_catalog or self.is_full_support
 
 
+class EngineRunMode(Enum):
+    SINGLE_MODE_ENGINE = 1
+    STANDALONE = 2
+    CLUSTER = 3
+    CLOUD = 4
+
+    @property
+    def is_single_mode_engine(self) -> bool:
```

Comment:
What's the difference between single mode engine and standalone?

Reply:
I was thinking that there are some engines that can run standalone or in a cluster-like configuration without any changes in behavior. SQLMesh likely wouldn't even know which way it's configured in, so it's not really accurate to say that it's "in" any particular mode.

So the idea is to distinguish engines whose behavior is constant regardless of mode from engines whose mode is relevant for sqlmesh behavior.

The mode enum isn't used outside clickhouse right now, though, so it kind of doesn't matter what value it's set to for other engines.

Reply:
The name "single mode engine" is definitely not consistent with my explanation, though.

If we keep that value at all, maybe change to `NO_MODE_BEHAVIOR_CHANGES` or similar?

Reply:
Hmm, it still feels like a Clickhouse abstraction leak. Is it actually needed at all outside the internals of the Clickhouse adapter? I didn't notice any checks for `engine_run_mode` in the evaluator

Reply:
no, it's just internal to the CH adapter